pkg/store/bucket.go

```diff
@@ -1229,6 +1230,13 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		defer s.queryGate.Done()
 	}
 
+	stream := srv.(grpc.ServerStream)
+	tenant, err := tenancy.GetTenantFromGRPCMetadata(stream.Context())
```

Comment:
We don't need to typecast here, `grpc.ServerStream` is an embedded struct in `storepb.Store_SeriesServer`, so we can call methods directly.
```suggestion
	tenant, err := tenancy.GetTenantFromGRPCMetadata(srv.Context())
```

Reply:
Done!